src/simsopt/field/coil.py

```diff
@@ -117,6 +154,17 @@ def vjp(self, v_current):
     def get_value(self):
         return self.current_a.get_value() + self.current_b.get_value()
 
+    def as_dict(self) -> dict:
+        d = {}
+        d["@module"] = self.__class__.__module__
+        d["@class"] = self.__class__.__name__
+        d["current"] = self.get_value()
+        return d
```

Comment:
I have merged this branch onto my permanent_magnets branch and this section of code appears not to work if ScaledCurrents are involved. To get it correct I needed to make the following changes to the CurrentSum class:

```
def __init__(self, current_a, current_b, **kwargs):
        self.current_a = current_a
        self.current_b = current_b
        sopp.CurrentBase.__init__(self)
        CurrentBase.__init__(self, x0=np.asarray([]), depends_on=[current_a, current_b], **kwargs)
       
def as_dict(self) -> dict:
        d = {}
        d["@module"] = self.__class__.__module__
        d["@class"] = self.__class__.__name__
        d["current_a"] = self.current_a.as_dict()
        d["current_b"] = self.current_b.as_dict()
        return d
@classmethod
    def from_dict(cls, d):
        decoder = MontyDecoder()
        current_a = decoder.process_decoded(d["current_a"])
        current_b = decoder.process_decoded(d["current_b"])
        return cls(current_a, current_b)
```